packages/sdk/src/wasm/VerifiablePresentationV1/audit-record.ts

```diff
@@ -101,6 +114,212 @@ export function create(
+        throw new Error('Mismatch in number of atomic statements for statement/claim');
+
+    verifyAtomicStatements(statement.statement, claims.statement);
+
+    if (isAccountClaims(claims)) {
+        await verifyAccountClaims(claims, statement, grpc);
+    } else {
+        verifyIdentityClaims(claims, statement);
+    }
+}
+
+async function verifyPresentationRequest(
+    verificationRequest: VerificationRequestV1.Type,
+    presentationRequest: VerifiablePresentationV1.Request,
+    grpc: ConcordiumGRPCClient
+) {
+    verificationRequest.credentialStatements.length === presentationRequest.subjectClaims.length ||
+        bail(
+            `Mismatch in number of statements/claims: ${verificationRequest.credentialStatements.length} request statements found, ${presentationRequest.subjectClaims.length} presentation claims found`
+        );
+
+    for (let i = 0; i < verificationRequest.credentialStatements.length; i++) {
+        await verifyClaims(verificationRequest.credentialStatements[i], presentationRequest.subjectClaims[i], grpc);
+    }
+}
+
+/**
+ * Creates a verification audit record after performing comprehensive validation of the presentation against the request.
+ *
+ * @param id - Unique identifier for the audit record
+ * @param request - The verification request containing statements to verify
+ * @param presentation - The verifiable presentation to validate
+ * @param grpc - Concordium gRPC client for on-chain verification
+ * @param network - Network identifier for verification context
+ * @param blockHash - Optional block hash to verify against a specific chain state
+ *
+ * @returns A verification result containing either the audit record on success or an error on failure
+ *
+ * @remarks
+ * This function performs four validation steps:
+ * 1. Compares contexts between request and presentation
+ * 2. Verifies cryptographic integrity of the presentation with public on-chain data
+ * 3. Checks that all credentials are active
+ * 4. Validates presentation claims against request statements
+ */
+export async function createChecked(
+    id: string,
+    request: VerificationRequestV1.Type,
+    presentation: VerifiablePresentationV1.Type,
+    grpc: ConcordiumGRPCClient,
+    network: Network,
+    blockHash?: BlockHash.Type
+): Promise<VerificationResult<VerificationAuditRecordV1>> {
+    // 1. check the context
+    try {
+        compareContexts(request, presentation, grpc);
+    } catch (e) {
+        return { type: 'failed', error: e as Error };
+    }
+
+    // 2. verify cryptographic integrity of presentation
+    const verification = await VerifiablePresentationV1.verifyWithNode(presentation, grpc, network, blockHash);
+    if (verification.type === 'failed') return verification;
+
+    // 3. Checck that none of the credentials have expired
+    verification.result.credentialsStatus.every((cs) => cs === CredentialStatus.Active) ||
+        bail('One or more credentials included in the presentation has expired.');
```

Comment:
```suggestion
        bail('One or more credentials included in the presentation is not active.');
```

Can also be `non-active/revoked`?